core/controllers/admin.py

```diff
@@ -848,3 +848,16 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class GetInteractionsByExplorationId(base.BaseHandler):
+    """This function returns a list of interaction id's for a exploration id."""
+
+    @acl_decorators.can_access_admin_page
+    def get(self):
+        exploartion_id = self.request.get('exploration_id')
+        exploration = exp_fetchers.get_exploration_by_id(exploartion_id)
+        interaction_ids = []
```

Comment:
Maybe use a set to begin with? I also think you can maybe do this in a more Pythonic away:

```python
interaction_ids = { state.interaction.id for state in exploration.states.items() }
```

Which uses set comprehension to collapse the for-loop and avoid the extra ``set()`` conversion you do on line 863.

Reply:
i tried this one but had some confusion about this so didn't move forward but will do this now

Reply:
@BenHenning Getting an error after using the above can you please help find the mistake
AttributeError: 'tuple' object has no attribute 'interaction'

Reply:
@BenHenning  ??

Reply:
@DubeySandeep can you help in this thing I'm not able why I'm not getting interaction inside state tuple

Reply:
Hey @aggarwalpulkit596 sorry for the delay.

I made a mistake in the logic. ``items()`` returns a tuple for the name and state. I suggest trying the following:

```python
interaction_ids = { state.interaction.id for state in exploration.states.values() }
```

Also in the future, when you run into these types of issues I suggest looking at similar code in the Oppia codebase to try and understand what was wrong with the code. In this case, I found the mistake by searching for ``states.items()`` which showed that ``states`` is actually a tuple of name and state: https://github.com/oppia/oppia/search?l=Python&q=states.items.

Reply:
Okay and for future, I will keep these points in mind :)
It's just that this is my first doing open source contribution in a python project and i haven't taken up any python projects as of now but I'm learning things in bits and pieces would get better over the time 
Thanks for your help @DubeySandeep  @BenHenning